src/envelope.rs

```diff
@@ -21,8 +21,10 @@ use std::str::FromStr;
  */
 #[derive(Debug)]
 pub struct SentryEnvelope {
-    pub raw_body: String,
+    pub raw_body: Vec<u8>,
     pub dsn: Dsn,
+    pub is_safe: bool,
+    pub user_ip: String,
```

Comment:
How about making this an Enum representing the various IP that we can have :
- an ip from x-forwarded-for
- the client IP